app/controllers/api/v2/bulk_snapshots_controller.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class BulkSnapshotsController < V2::BaseController
+      api :POST, '/snapshots/bulk/create_snapshot', N_('Create snapshots for multiple hosts')
+      param :host_ids, Array, of: Integer, desc: N_('Array of host IDs to create snapshots for'), required: true
+      param :snapshot, Hash, desc: N_('Snapshot parameters'), required: true do
+        param :name, String, desc: N_('Name of the snapshot'), required: true
+        param :description, String, desc: N_('Description of the snapshot'), required: false
+      end
+      param :include_ram, [true, false], desc: N_('Include RAM in the snapshot'), required: false
```

Comment:
I would prefer, to have the footprint of the method to be something like:

param :mode, [:full, :quiesce, :include_ram]

because its not possible to have quiesce and include_ram to the same time.